src/Avalonia.Controls/ManagedPointer.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Reactive.Linq;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using Avalonia.Input.Raw;
+using Avalonia.Media;
+using Avalonia.Threading;
+
+namespace Avalonia.Controls
+{
+    public class ManagedPointer : TemplatedControl
+    {
+        private class PointerTransform : ITransform
+        {
+            public Point Position { get; set; }
+
+            public Matrix Value => Matrix.CreateTranslation(Position.X, Position.Y);
+        }
+
+        public ManagedPointer(TopLevel visualRoot)
+        {
+            IsHitTestVisible = false;
+            ZIndex = int.MaxValue;
+
+            RenderTransform = new PointerTransform();
+
+            var layer = OverlayLayer.GetOverlayLayer(visualRoot);
+
+            ((ISetLogicalParent)this).SetParent(visualRoot);
+            layer.Children.Add(this);
+
+            InputManager.Instance.PreProcess.OfType<RawPointerEventArgs>().Subscribe(x =>
+            {
+                IsVisible = true;
+                
+                if (x is { Type: RawPointerEventType.Move } args &&
+                    RenderTransform is PointerTransform t)
+                {
+                    t.Position = args.Position;
+                    InvalidateVisual();
+                }
+            });
+
+            InputManager.Instance.PreProcess.OfType<RawPointerEventArgs>()
```

Comment:
This timer should be started on attaching to the visual tree (or logical?), and stopped on detaching. 
Or it should be even optional, somebody might prefer always visible pointer.

Reply:
why attach to visual tree?

Can add a way to turn that on or off.